src/probnum/linops/_arithmetic_fallbacks.py

```diff
@@ -227,6 +223,9 @@ def __repr__(self):
             res += f"\t{s}, \n"
         return res + "]"
 
+    def _solve(self, B: np.ndarray) -> np.ndarray:
+        return functools.reduce(lambda vec, op: op @ vec, self._factors, B)
```

Comment:
I may be misunderstanding, but doesn't this line compute $(B F_1) F_2$, where the linear operator is given by $L = F_1 F_2$? Shouldn't it rather be something like 
```suggestion
        return functools.reduce(lambda rhs, op: op.solve(rhs), self._factors, B)
```
If yes, also add a test for the product linear operator.

Reply:
Oops, that was kind of a last minute addition. I wonder why codecov did not catch the missing coverage?